eva-server/src/main/java/uk/ac/ebi/eva/server/ws/VariantWSServer.java

```diff
@@ -164,27 +169,34 @@ public QueryResponse checkVariantExists(@PathVariable("variantId") String varian
         String invalidCoordinatesMessage =
                 "Invalid position and alleles combination, please use chr:pos:ref or chr:pos:ref:alt";
 
-        if (variantId.contains(":")) {
-            String[] regionId = variantId.split(":", -1);
-            if (regionId.length < 3) {
-                response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
-                return setErrorQueryResponse(invalidCoordinatesMessage);
-            }
+        try {
+            if (variantId.contains(":")) {
+                String[] regionId = variantId.split(":", -1);
+                if (regionId.length < 3) {
+                    response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+                    return setErrorQueryResponse(invalidCoordinatesMessage);
+                }
 
-            String alternate = (regionId.length > 3) ? regionId[3] : null;
+                String alternate = (regionId.length > 3) ? regionId[3] : null;
 
-            if (studies != null && !studies.isEmpty()) {
-                variantEntities = queryByCoordinatesAndAllelesAndStudyIds(regionId[0], Integer.parseInt(regionId[1]),
+
+                if (studies != null && !studies.isEmpty()) {
+                    variantEntities = queryByCoordinatesAndAllelesAndStudyIds(regionId[0], Integer.parseInt(regionId[1]),
                         regionId[2], alternate, studies);
-            } else {
-                variantEntities = queryByCoordinatesAndAlleles(regionId[0], Integer.parseInt(regionId[1]), regionId[2],
+                } else {
+                    variantEntities = queryByCoordinatesAndAlleles(regionId[0], Integer.parseInt(regionId[1]), regionId[2],
                                                                alternate, null, null);
-            }
+                }
 
-        } else {
-            List<VariantRepositoryFilter> filters = new FilterBuilder().withStudies(studies).build();
-            variantEntities = service.findByIdsAndComplexFilters(variantId, filters, null, null,
+            } else {
+                List<VariantRepositoryFilter> filters = new FilterBuilder().withStudies(studies).build();
```

Comment:
Please check, but I think only this branch can throw. If so, it would be preferable to surround only that part with a try-catch block, because that way it is clear for the person reading the code that the rest doesn't throw.

Reply:
@cyenyxe Actually, the exception is thrown by 3 branches: line 184 (queryByCoordinatesAndAllelesAndStudyIds), line 187 (queryByCoordinatesAndAlleles), and line 193 (findByIdsAndComplexFilters)... so it's better to leave the try-catch block like it's now? Or maybe use 3 try-catch blocks?

Reply:
Okay, I didn't remember those would also throw. Please let it as is then, it will be more clear than having a lot of try-batch blocks.